sparseSpACE/PerformTestCase.py

```diff
@@ -115,148 +179,272 @@ def performTestcaseArbitraryDim(f, a, b, adaptiveAlgorithmVector, maxtol, dim, m
+                csv_out.writerow(["error"] + error)
+                csv_out.writerow([])
+
+    def execute_standard_combi(self, grid: Grid, lmin: int, max_lmax: int, dim: int, reference_solution: Sequence[float],
+                               evaluation_points: Sequence[Tuple[float, ...]]):
+        # calculate standard combination scheme results
+        errorArrayStandard = []
+        pointArray = []
+        distinctFEvalArray = []
+        operation = Integration(self.f, grid, dim, reference_solution)
+        standardCombi = StandardCombi(self.a, self.b, operation=operation)
+        interpolation_errorL2 = []
+        interpolation_errorMax = []
+
+        for i in range(lmin + 1, lmin + max_lmax):
+            scheme, error, result = standardCombi.perform_operation(lmin, i)
+            errorArrayStandard.append(error / abs(reference_solution))
+            pointArray.append(standardCombi.get_total_num_points())
+            distinctFEvalArray.append(standardCombi.get_total_num_points(distinct_function_evals=True))
+
+            if evaluation_points is not None:
+                interpolated_values = np.asarray(standardCombi(evaluation_points))
+                real_values = np.asarray([self.f.eval(point) for point in evaluation_points])
+                diff = [real_values[i] - interpolated_values[i] for i in range(len(evaluation_points))]
+                # print(interpolated_values, diff)
+                interpolation_errorL2.append(scipy.linalg.norm(diff, 2))
+                interpolation_errorMax.append(scipy.linalg.norm(diff, np.inf))
+
+        return pointArray, distinctFEvalArray, errorArrayStandard, interpolation_errorL2, interpolation_errorMax
+
+
+class PlotTestCase:
+    def __init__(self, colormap: dict, line_style_map: dict, import_filepath="../", export_filepath="./"):
+        """
+        Initializes the plots
+
+        Parameters
+        ----------
+        colormap: dict that maps algorithm name strings to color strings, e.g.:
+                    colormap = {
+                        "Gauss-Legendre Grid (Standard Combi) lmin=1": "tab:gray",
+                        "Trapezoidal Grid (Standard Combi) lmin=1": "violet",
+                        "Trapezoidal Grid": "red",
+                        "Trapezoidal Grid (Rebalancing)": "royalblue",
+                    }
+        line_style_map: dict that maps algorithm name strings to color strings, e.g.:
+                        line_style_map = {
+                            "Gauss-Legendre Grid (Standard Combi) lmin=1": "--",
+                            "Trapezoidal Grid (Standard Combi) lmin=1": "--",
+                            "Trapezoidal Grid": "-.",
+                            "Trapezoidal Grid (Rebalancing)": "-.",
+                        }
+        import_filepath: From where the CSV files are imported
+        export_filepath: Where the plots are written
+        """
+        self.colormap = colormap
+        self.line_style_map = line_style_map
```

Comment:
It would be nice to have default colormaps and linepaths. It is tedious to set them always manually. Maybe instead pass the algorithms and this constructor sets some default colours and line styles to them? Or always assign colors anew for each plot?